addons/mail/models/mail_activity.py

```diff
@@ -127,9 +127,27 @@ def default_get(self, fields):
         'Document Name', compute='_compute_res_name', compute_sudo=True, store=True,
         help="Display name of the related document.", readonly=True)
     # activity
+    @api.model
+    def default_get_activity_type(self):
+        activity_type = self.env["mail.activity.type"]
+        ir_model_obj = self.env['ir.model']
+        current_model_name = self.env.context['default_res_model']
+        current_model = ir_model_obj.search([('model', '=', current_model_name)])
+        current_model_id = current_model[0].id
+        todo = activity_type.search([('name','ilike','to do')], limit=1)
```

Comment:
It's a trap :) !

Name is translatable, meaning if in french it is called "A faire", your search will not work as name depends on the current user's language.

Here Todo refers to the data coming with mail module, you can use self.env.ref('mail.mail_activity_data_todo', raise_if_not_found=False). You receives the record or None.